### PR TITLE
feat: add script create release with submodules

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

* Create version update in the monorepo and submodule.
* Updated the submodule repo with commit and tag. 
* Updated the main repo with commit and tag.

### Why was this change needed?

So we could create a version in one flow.

### Other information (Screenshots)

Addition point to check:
* We need to validate that there is no uncommit files in a submodule, if there is we need to throw an error (in the main repo lerna takes care of it)

### CODE REVIEW

File: scripts/release-with-submodule.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  console.log(`Running command: ${command}${directory ? `directory:${directory}` : '.'}`);
+  execSync(command, {
+    stdio: 'inherit',
+    cwd: directory,
+  });
+}
+
+try {
+  console.log('Step 1: Run lerna version in monorepo without committing or creating a release');
+  runCommand('pnpm lerna version patch --yes --no-git-tag-version --no-push --force-publish');
```
Comment (djabarovgeorge) [inline scripts/release-with-submodule.ts:17]:
```suggestion
  runCommand('pnpm lerna version patch --yes --no-git-tag-version --no-push ');
```

needed to be removed, i added it only for local testing.

File: scripts/release-with-submodule.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    runCommand('git commit -m "chore: update versions"', submoduleDir);
+    runCommand('git push', submoduleDir);
+
+    console.log('Step 6: Create release tag for submodule');
+    runCommand(`git tag v${newVersion}`, submoduleDir);
+    runCommand('git push --tags', submoduleDir);
+
+    console.log('Step 7: Commit version changes and submodule updated hash main repository');
+    runCommand('git add .');
+    runCommand('git commit -m "chore: update versions"');
```
Comment (djabarovgeorge) [inline scripts/release-with-submodule.ts:70]: @scopsydo Do you know if we have some commit convention, that will require an update in the commit messages?

Comment (LetItRock) [reply scripts/release-with-submodule.ts:70]: @djabarovgeorge Usually the message is like this

Screenshot 2023-10-18 at 12 51 02

**Review by LetItRock — APPROVED** [review]

File: scripts/release-with-submodule.ts (hunk 2 above)
Comment (djabarovgeorge) [reply scripts/release-with-submodule.ts:70]: totally right that's why I asked, maybe now we would like to customize it abit.

Comment (djabarovgeorge) [reply scripts/release-with-submodule.ts:70]: will update to 
`chore(release): publish`

Comment (LetItRock) [reply scripts/release-with-submodule.ts]: shouldn't we keep `- ci skip`?

Comment (djabarovgeorge) [reply scripts/release-with-submodule.ts]: If you say so, i was not sure what was it purpose 😕

**Pull request merged**